### DISCUSSION TITLE
How can I add a vercel env variable?

### DISCUSSION THREAD

**Discussion by Raul-Rauqoz** [vercel/community#5261] [Help]
### Question

sorry for this noobie question but I'm checking if can I add a new env variable using gha 
but I don't understand how I need to send the [name] on:
`vercel env add [name]`
If I try for example 
`vercel env add test preview` or `vercel env add test=value preview`
the output is: 
```
Vercel CLI 33.0.1
Retrieving project…
? What’s the value of test? 
```
Or do I need another way?

thanks!

**Comment by amyegan** [comment]
Hey Raul! You can use [`vercel env add [name] [environment] < [file]`](https://vercel.com/docs/cli/env#extended-usage) to add an environment variable and set it's value to the contents of a local file all in one step. Otherwise, you will get a prompt to provide the value as a second step like you saw. That works best on local in situations where you don't change the value very often.

If you're changing the value with each deployment, I recommend a different approach. The [`vercel deploy`](https://vercel.com/docs/cli/deploy#) command has the [`--build-env`](https://vercel.com/docs/cli/deploy#build-env) and [`--env`](https://vercel.com/docs/cli/deploy#env) options to provide environment variables for a single deployment.
